### DISCUSSION TITLE
关于ppdns本身是否应该被分流

### DISCUSSION THREAD

**Discussion by wuiiled** [kkkgo/PaoPaoDNS#227] [问答和帮助]
在 #106 中看到作者提到网络要求ppdns本身不能被分流，不能被劫持DNS，正常的话Non-CN RAW-IP URL和IP INFO是你的宽带IP信息

但是在ppgw中看到作者说如果ppdns后面接fakeip网关后，Non-CN RAW-IP URL和IP INFO是代理ip信息

我现在的网络是ppdns--mihomo，如果在mihomo中加一个规则让来自ppdns流量全部直连，可以做到Non-CN RAW-IP URL和IP INFO是宽带IP信息，如果不加这条规则的话Non-CN RAW-IP URL和IP INFO是代理ip信息，两种设置方法运行test.sh都可以pass，请问这两种方式有什么区别？用哪种更好

**Comment by kkkgo** [comment]
调试脚本是用来帮你定位问题的，不能颠倒因果关系，就像为跑分优化一样。
没有被分流的情况下Non-CN RAW-IP URL和IP INFO都是应该是正常的宽带IP。
跟fakeip网关也没有关系，Non-CN RAW-IP URL就是不使用域名的检测。IP INFO的检测域名默认在force_dnscrypt_list.txt，也不会被fakeip，你可能是哪里搞错了。
IP分流/劫持会导致递归DNS不准确。此外不需要被代理的数据也没必要再走一圈碰到规则再回去。

**Reply by wuiiled** [reply]
> 调试脚本是用来帮你定位问题的，不能颠倒因果关系，就像为跑分优化一样。 没有被分流的情况下Non-CN RAW-IP URL和IP INFO都是应该是正常的宽带IP。 跟fakeip网关也没有关系，Non-CN RAW-IP URL就是不使用域名的检测。IP INFO的检测域名默认在force_dnscrypt_list.txt，也不会被fakeip，你可能是哪里搞错了。 IP分流/劫持会导致递归DNS不准确。此外不需要被代理的数据也没必要再走一圈碰到规则再回去。

也就是说ppdns运行的理想状态是直连状态。
想探究这个主要是有过一些域名会返回0.0.0.0的解析结果，在 #83 里看到了解决办法，但是把ppdns直连之后又出现了pppref跑域名时成功率只有60%左右的问题（之前跑完全部列表是95%左右），但是实际使用也没有发现有问题，很疑惑

还有一个问题就是我环境变量中设置了socks5，但是debug中5303测试sock5显示timeout，这个是正常的吗？

**Reply by kkkgo** [reply]
paopao-pref判定的是有返回结果算成功，不关心结果真假或者fakeip。成功率与网络环境和机器性能相关。
5303没有结果说明设置的SOCKS5变量不正确或者不可用。
